nautobot/extras/tests/test_views.py

```diff
@@ -988,28 +1036,234 @@ def setUpTestData(cls):
             approval_required=True,
             start_time=datetime.now(),
         )
+        ScheduledJob.objects.create(
+            name="test4",
+            task="nautobot.extras.jobs.scheduled_job_handler",
+            job_class="-",
+            interval=JobExecutionType.TYPE_IMMEDIATELY,
+            user=user,
+            approval_required=False,
+            start_time=datetime.now(),
+        )
+
+    def test_only_approvable_is_listed(self):
+        self.add_permissions("extras.view_scheduledjob")
+
+        response = self.client.get(self._get_url("list"))
+        self.assertHttpStatus(response, 200)
+        self.assertNotIn("test4", extract_page_body(response.content.decode(response.charset)))
 
 
 class JobResultTestCase(
+    TestJobMixin,
+    ViewTestCases.GetObjectViewTestCase,
     ViewTestCases.ListObjectsViewTestCase,
+    ViewTestCases.DeleteObjectViewTestCase,
+    ViewTestCases.BulkDeleteObjectsViewTestCase,
 ):
     model = JobResult
 
     @classmethod
     def setUpTestData(cls):
         obj_type = ContentType.objects.get(app_label="extras", model="job")
         JobResult.objects.create(
-            name="test1",
+            name="local/test_view/TestJob",
             job_id=uuid.uuid4(),
             obj_type=obj_type,
         )
         JobResult.objects.create(
-            name="test2",
+            name="local/test_view/TestJob2",
             job_id=uuid.uuid4(),
             obj_type=obj_type,
         )
         JobResult.objects.create(
-            name="test3",
+            name="local/test_view/TestJob3",
             job_id=uuid.uuid4(),
             obj_type=obj_type,
         )
+
+
+class JobTestCase(
+    TestCase,
+):
+    @override_settings(EXEMPT_VIEW_PERMISSIONS=[])
+    def test_list_without_permission(self):
+        self.assertHttpStatus(self.client.get(reverse("extras:job_list")), 403)
+
+    @override_settings(EXEMPT_VIEW_PERMISSIONS=["*"])
+    def test_list(self):
+        response = self.client.get(reverse("extras:job_list"))
+        self.assertHttpStatus(response, 200)
+
+        response_body = extract_page_body(response.content.decode(response.charset))
+        self.assertIn("DummyJob", response_body)
```

Comment:
Does this test need to be skipped if the dummy plugin isn't installed?

I'm also surprised that this one doesn't need `JOBS_ROOT=DUMMY_JOBS` - what am I missing here?

Reply:
Good question! The test seems to work for me; I think the dummy plugin is always installed during testing (though I don’t know where that happens)?

Reply:
Really our tests should always be run with the dummy plugin installed since we would have lower test coverage without it - but there is an existing pattern for skipping tests just in case it somehow isn't installed - see `nautobot/extras/tests/test_plugins.py` and `test_api.py` for examples. 

To your question, `nautobot/core/tests/nautobot_config.py` adds "dummy_plugin" to `PLUGINS`, and the plugin is installed by Poetry since it's listed as a dev-dependency in `pyproject.toml` - so these days it *should* always be available for tests, I think.

Reply:
Thanks for the context! I changed this to use one of the jobs from `dummy_jobs` instead, such that we don’t rely on the plugin being present where we don’t have to :)